### PR TITLE
feat: set path normalization settings

### PR DESCRIPTION
**Pull request by tanujd11**
**What type of PR is this?**
 Feature to set defaults in http connection manager

**What this PR does / why we need it**:
Sets path normalization settings according to the best practices https://www.envoyproxy.io/docs/envoy/latest/configuration/best_practices/edge

**Which issue(s) this PR fixes**:

Fixes part of #1048

### CODE REVIEW

**Comment by tanujd11** [comment]
@Xunzhuo could you please rerun the CI. I guess there was some issue in uploading the Codecov report in the last run. Thanks

File: internal/xds/translator/listener.go [hunk 1]
```diff
@@ -101,6 +101,11 @@ func (t *Translator) addXdsHTTPFilterChain(xdsListener *listenerv3.Listener, irL
 			Name:       wellknown.Router,
 			ConfigType: &hcmv3.HttpFilter_TypedConfig{TypedConfig: routerAny},
 		}},
+		// normalize paths according to RFC 3986
+		NormalizePath: &wrapperspb.BoolValue{Value: true},
+		// merge adjacent slashes in the path
+		MergeSlashes:                 true,
+		PathWithEscapedSlashesAction: hcmv3.HttpConnectionManager_UNESCAPE_AND_REDIRECT,
```
Comment (arkodg) [inline internal/xds/translator/listener.go:108]: this one is tricky
```
UNESCAPE_AND_REDIRECT if downstream client supports HTTP redirect (i.e. a browser). This option minimizes possibility of path confusion by forcing request to be re-issued with the same path across all parties: downstream client, Envoy and upstream server. Note that gRPC requests will still be rejected with the INTERNAL (13) error code, as gRPC clients do not support redirect.
```
so ideally it should be enabled for HTTP and disabled for gRPC.
Can this one be rm'd from this PR and solved in a follow up

Comment (tanujd11) [reply internal/xds/translator/listener.go:108]: So the default best practice for gRPC is to REJECT_REQUEST 

```REJECT_REQUEST if dowstream clients are expected to use [RFC 3986](https://www.ietf.org/rfc/rfc3986.txt) compliant normalized paths (i.e. gRPC clients).```

And the UNESCAPE_AND_REDIRECT will also reject the request right? So won't this cover both the http and gRPC? We can take this in next iteration but is there any gap in my understanding here?

**Review by arkodg — APPROVED** [review]
🚀

**Review by AliceProxy — APPROVED** [review]
lgtm, thanks!

**Pull request merged**